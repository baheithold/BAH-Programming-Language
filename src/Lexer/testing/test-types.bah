# This file tests valid types


# Integers
1234567890
0
-1
-2
-100
-5000
-1234567890

# Reals
0.0
3.14159
123.456789
-0.1
-5.123
-3.14159
-.05
-5.

# Strings
"This is a standard string"
"   some spaces before"
"some spaces after  "
"some    random spaces   "
"\"this is a quote\""
"and\\or"

# Booleans
true
false

# Null
null
