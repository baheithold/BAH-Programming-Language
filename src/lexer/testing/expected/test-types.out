INTEGER_TYPE 1234567890
INTEGER_TYPE 0
MINUS_BINARY
INTEGER_TYPE 1
MINUS_BINARY
INTEGER_TYPE 2
MINUS_BINARY
INTEGER_TYPE 100
MINUS_BINARY
INTEGER_TYPE 5000
MINUS_BINARY
INTEGER_TYPE 1234567890
REAL_TYPE 0.000000
REAL_TYPE 3.141590
REAL_TYPE 123.456789
MINUS_BINARY
REAL_TYPE 0.100000
MINUS_BINARY
REAL_TYPE 5.123000
MINUS_BINARY
REAL_TYPE 3.141590
MINUS_BINARY
REAL_TYPE 0.050000
MINUS_BINARY
REAL_TYPE 5.000000
STRING_TYPE "This is a standard string"
STRING_TYPE "   some spaces before"
STRING_TYPE "some spaces after  "
STRING_TYPE "some    random spaces   "
STRING_TYPE ""this is a quote""
STRING_TYPE "and\or"
BOOLEAN_TYPE
BOOLEAN_TYPE
NULL_TYPE
END_OF_INPUT
